cypress/integration/settings/circulation/general/edit-circulation-rules.spec.js

```diff
@@ -0,0 +1,155 @@
+  const calloutMessages = {
+    CIRCULATION_RULES_UPDATE_SUCCESS: 'Rules were successfully updated.',
+  };
+  const newNoticePolicyData = {
+    name: getTestEntityValue(),
+    description: 'description',
+    active: true,
+    id: uuid(),
+  };
+  const newLoanPolicyData = {
+    id: uuid(),
+    name: getTestEntityValue(),
+    description: 'description',
+    loanable: false,
+    renewable: false,
+  };
+  const newMaterialTypeData = {
+    source: 'local',
+    name: getTestEntityValue(),
+    id: uuid(),
+  };
+  const newOverdueFinePolicyData = {
+    countClosed: true,
+    forgiveOverdueFine: true,
+    gracePeriodRecall: true,
+    maxOverdueFine: '0.00',
+    maxOverdueRecallFine: '0.00',
+    name: getTestEntityValue(),
+    description: 'description',
+    id: uuid(),
+  };
+  const newRequestPolicyData = {
+    requestTypes: [
+      'Hold',
+    ],
+    name: getTestEntityValue(),
+    description: 'description',
+    id: uuid(),
+  };
+  const newLostItemFeePolicyData = {
+    chargeAmountItem: {
+      amount: '0.00',
+      chargeType: 'anotherCost',
+    },
+    lostItemProcessingFee: '0.00',
+    chargeAmountItemPatron: false,
+    chargeAmountItemSystem: false,
+    returnedLostItemProcessingFee: false,
+    replacedLostItemProcessingFee: false,
+    replacementProcessingFee: '0.00',
+    replacementAllowed: false,
+    lostItemReturned: 'Charge',
+    name: getTestEntityValue(),
+    description: 'description',
+    lostItemChargeFeeFine: {
+      duration: 1,
+      intervalId: 'Days',
+    },
+    id: uuid(),
+  };
+  let originalCirculationRules;
+  let newUserId;
+
+  beforeEach(() => {
+    cy.createTempUser().then(({
+      username,
+      password,
+      userId,
+    }) => {
+      newUserId = userId;
+
+      cy.getAdminToken();
+
+      cy.getCirculationRulesApi().then((circulationRules) => {
+        originalCirculationRules = circulationRules;
+      });
+      cy.createMaterialTypeApi(newMaterialTypeData);
+      cy.createNoticePolicyApi(newNoticePolicyData);
+      cy.createLoanPolicyApi(newLoanPolicyData);
+      cy.createRequestPolicyApi(newRequestPolicyData);
+      cy.createLostItemFeePolicyApi(newLostItemFeePolicyData);
+      cy.createOverdueFinePolicyApi(newOverdueFinePolicyData);
+
+      cy.login(username, password);
+    });
+
+    cy.visit(SettingsMenu.circulationRulesPath);
+  });
+
+  afterEach(() => {
+    cy.editCirculationRulesApi(originalCirculationRules);
+    cy.deleteMaterialTypeApi(newMaterialTypeData.id);
+    cy.deleteNoticePolicyApi(newNoticePolicyData.id);
+    cy.deleteLoanPolicyApi(newLoanPolicyData.id);
+    cy.deleteRequestPolicyApi(newRequestPolicyData.id);
+    cy.deleteLostItemFeePolicyApi(newLostItemFeePolicyData.id);
+    cy.deleteOverdueFinePolicyApi(newOverdueFinePolicyData.id);
+    cy.deleteUser(newUserId);
+  });
+
+  it('C2268: Add notice policy to circulation rules', { tags: [TestType.smoke] }, () => {
+    cy.do([
+      CodeMirror().clear(),
```

Comment:
redesign to wrappers related with folio fragments. All interactions with web elements should be wrapped and encapsulated into fragments

Reply:
done